src/structure/pdb.jl

```diff
@@ -30,26 +49,326 @@ end
+                # Check PDB ID is 4 characters long and only consits of alphanumeric characters
+                if length(pdbid) != 4 || ismatch(r"[^a-zA-Z0-9]", pdbid)
+                    throw(ArgumentError("Not a valid PDB ID: \"$pdbid\""))
+                end
+                push!(pdbidlist,pdbid)
+            end
+            linecount +=1
+        end
+    end
+    rm(tempfilepath)
+    return pdbidlist
+end
+
+
+"""
+Returns a list of PDB codes in the weekly PDB status file from the given URL;
+Argument `url` must specify the URL of any weekly status file from RCSB Server.
+"""
+function pdbstatuslist(url::AbstractString)
+    statuslist = String[]
+    filename = split(url,"/")[end]
+    info("Fetching weekly status file $filename from RCSB Server...")
+    tempfilepath = tempname()
+    download(url, tempfilepath)
+    open(tempfilepath) do input
+        for line in eachline(input)
+            # The first 4 characters in the line is the PDB ID
+            pdbid = uppercase(line[1:4])
+            # Check PDB ID is 4 characters long and only consits of alphanumeric characters
+            if length(pdbid) != 4 || ismatch(r"[^a-zA-Z0-9]", pdbid)
+                throw(ArgumentError("Not a valid PDB ID: \"$pdbid\""))
+            end
+            push!(statuslist,pdbid)
+        end
+    end
+    rm(tempfilepath)
+    return statuslist
+end
+
+
+"""
+Returns three lists of the newest weekly files (added,modified,obsolete)
+from RCSB PDB Server.
+"""
+function pdbrecentchanges()
+    addedlist = pdbstatuslist("ftp://ftp.wwpdb.org/pub/pdb/data/status/latest/added.pdb")
+    modifiedlist = pdbstatuslist("ftp://ftp.wwpdb.org/pub/pdb/data/status/latest/modified.pdb")  
+    obsoletelist = pdbstatuslist("ftp://ftp.wwpdb.org/pub/pdb/data/status/latest/obsolete.pdb")  
+    return addedlist, modifiedlist, obsoletelist
+end
+
+
+"""
+Returns a list of all obsolete entries ever in the RCSB PDB server.
+"""
+function pdbobsoletelist()
+    obsoletelist = String[]
+    info("Fetching list of all obsolete PDB Entries from RCSB PDB Server...")
+    tempfilepath = tempname()
+    download("ftp://ftp.wwpdb.org/pub/pdb/data/status/obsolete.dat", tempfilepath)
+    open(tempfilepath) do input
+        for line in eachline(input)
+            # Check if its an obsolete pdb entry and not headers
+            if line[1:6] == "OBSLTE"
+                # The 21st to 24th characters in obsolete pdb entry has the pdb id
+                pdbid = uppercase(line[21:24])
+                # Check PDB ID is 4 characters long and only consits of alphanumeric characters
+                if length(pdbid) != 4 || ismatch(r"[^a-zA-Z0-9]", pdbid)
+                    throw(ArgumentError("Not a valid PDB ID: \"$pdbid\""))
+                end
+                push!(obsoletelist,pdbid)
+            end
+        end
+    end
+    rm(tempfilepath)
+    return obsoletelist
+end
+
+
+"""
+Download a PDB file or biological assembly from the RCSB PDB server. 
```

Comment:
The first line should be the signature of the function (see: https://docs.julialang.org/en/stable/manual/documentation).